### PR TITLE
Kerberos smart card logon support

### PR DESCRIPTION
**Pull request by TheBestTvarynka**
Hi,
In this pull request, I've implemented the Kerberos smart card logon. General overview:
At the current moment, it uses the WinScard API to communicate with the smart card. It also means that this feature is working only on Windows machines (but it'll be fixed in the future 😉 ).

I tested it on the following scenarios:

* Win 11 -> Win 11 using `mstsc`.
* Win 11 -> Win 11 using `FreeRDP`.

Our QA will check all other cases.

More information you can read in those four pull requests description:

* https://github.com/Devolutions/sspi-rs/pull/143
* https://github.com/Devolutions/sspi-rs/pull/144
* https://github.com/Devolutions/sspi-rs/pull/145
* https://github.com/Devolutions/sspi-rs/pull/147

You can watch demo videos here: https://github.com/Devolutions/sspi-rs/pull/147#issue-1871359978.

Docs & references:

* [RFC 4556: Public Key Cryptography for Initial Authentication in Kerberos (PKINIT) (rfc-editor.org)](https://www.rfc-editor.org/rfc/rfc4556.html)
* [[MS-CSSP]: TSCredentials | Microsoft Learn](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-cssp/94a1ab00-5500-42fd-8d3d-7a84e6c2cf03)
* [Winscard.h header - Win32 apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/win32/api/winscard/)
* [Microsoft Base Smart Card Cryptographic Service Provider | Microsoft Learn](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/secsmart/microsoft-base-smart-card-cryptographic-service-provider)
* [`pcsc` crate](https://docs.rs/pcsc/latest/pcsc)
* [MS-CSSP: TSCredentials](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-cssp/94a1ab00-5500-42fd-8d3d-7a84e6c2cf03).
* [CredUnmarshalCredentialW](https://learn.microsoft.com/en-us/windows/win32/api/wincred/nf-wincred-credunmarshalcredentialw).

### CODE REVIEW

**Comment by CBenoit** [comment]
> Our QA will check all other cases.

Should we wait for your QA report before we land this?

**Comment by TheBestTvarynka** [comment]
Nope, we can merge it. I usually build and publish a test request after the merge. I'll create a separate PR with the fix if they find something

**Review by CBenoit — APPROVED** [review]
Sounds good! We’ll publish the crate once tests are done :+1:

**Pull request merged**
